test/acceptance/client.js

```diff
@@ -341,13 +408,65 @@ describe('The node.js OC client', function(){
         });
 
         it('should contain the error details', function(){
-          expect(error.toString()).to.match(/Error\: Server-side rendering failed: Error: (.*?) when connecting to http\:\/\/localhost\:1234 \(500\)/g);
+          var exp = getRegExpFromJson(expectedRequest),
+              expected = new RegExp('Error: Server-side rendering failed: request ' + exp + ' failed \\(Error: connect ECONNREFUSED(.*?)\\)');
+
+          expect(error.toString()).to.match(expected);
         });
       });
     });
 
     describe('when server-side rendering an existing component linked to a responsive registry', function(){
 
+      describe('when the component times-out', function(){
+
+        var error, result;
+
+        var expectedRequest = {
+          url: 'http://localhost:3030',
+          method: 'post',
+          headers: { 
+            accept: 'application/vnd.oc.unrendered+json',
+            'user-agent': 'oc-client-(.*?)'
+          },
+          timeout: 0.01,
+          json: true,
+          body: {
+            components: [{
+              name: 'errors-component',
+              parameters: {
+                errorType: 'timeout',
+                timeout: 1000
+              }
+            }]
+          }
+        };
+
+        before(function(done){
+          client.renderComponent('errors-component', {
+            parameters: { errorType: 'timeout', timeout: 1000 },
+            timeout: 0.01,
+            disableFailoverRendering: true
+          }, function(err, html){
+            error = err;
+            result = html;
+            done();
+          });
+        });
+
+        it('should contain a blank html response', function(){
+          expect(result).to.eql('');
+        });
+
+        it('should contain the error details', function(){
+
+          var exp = getRegExpFromJson(expectedRequest),
+              expected = new RegExp('Error: Server-side rendering failed: request ' + exp + ' failed \\(timeout\\)');
```

Comment:
It is there. The client has a [default overridable timeout value](https://github.com/opentable/oc/blob/master/client/src/sanitiser.js#L44) that is always passed through the request call. Now we log all the arguments that we pass to that function call ( a JSON.stringify of `expectedRequest` - a couple of lines up) - which contains timeout, headers, body payload, everything.

An example of an error (I censored some private data stuff):

[Error: Server-side rendering failed: request {"url":"http://xxx","method":"post","headers":{"accept-language":"fr-CA, fr, en, *","ot-requestid":"xxx","ot-referringhost":"127.0.0.1","ot-referringservice":"xxx","accept":"application/vnd.oc.unrendered+json","user-agent":"oc-client-1.10.0/v4.4.4-darwin-x64"},"timeout":5,"json":true,"body":{"components":[{"name":"xxx","parameters":{xxx},"version":"1.2.0"}]}} failed (timeout)] ]